sklearn/feature_extraction/text.py

```diff
@@ -864,6 +876,10 @@ def transform(self, raw_documents):
         X : sparse matrix, [n_samples, n_features]
             Document-term matrix.
         """
+        if isinstance(raw_documents, str):
+            raise ValueError(
+                "iterable over raw text documents expected, string object given")
```

Comment:
Oh sorry! I actually wanted it to be received but forgot to change that one. Tests incoming too.